src/js/angular/core/services/theme-service.js

```diff
@@ -0,0 +1,190 @@
+     * PluginRegistry.
+     */
+    const getSelectedThemeDefinition = () => {
+        const workbenchTheme = workbenchSettingsStorageService.getTheme();
+        return getThemeDefinitionByName(workbenchTheme);
+    };
+
+    /**
+     * @return {{name: string, label: string}} The selected theme or the default theme name and label.
+     */
+    const getTheme = () => {
+        let theme;
+        const selectedThemeDefinition = getSelectedThemeDefinition();
+        if (selectedThemeDefinition) {
+            theme = {
+                name: selectedThemeDefinition.name,
+                label: $translate.instant(selectedThemeDefinition.label) || selectedThemeDefinition.label
+            };
+        } else {
+            theme = getDefaultTheme();
+        }
+        return theme;
+    };
+
+    /**
+     * @return {{name: string, label: string}[]} An array with all registered theme names and labels.
+     */
+    const getThemes = () => {
+        const themeDefinitions = PluginRegistry.get(THEMES_EXTENSION);
+        return themeDefinitions.map((themeDefinition) => ({
+            name: themeDefinition.name,
+            label: $translate.instant(themeDefinition.label) || themeDefinition.label
+        }));
+    };
+
+    /**
+     * Gets a theme definition. If there is a saved theme, then theme definition with that name loaded from
+     * PluginRegistry and returned. Otherwise the default theme definition is returned.
+     * @return {Object}
+     */
+    const getThemeDefinition = () => {
+        const workbenchTheme = workbenchSettingsStorageService.getTheme();
+        let selectedThemeDefinition = PluginRegistry.findPlugin(THEMES_EXTENSION, (themeDefinition) => themeDefinition.name === workbenchTheme);
+        if (!selectedThemeDefinition) {
+            selectedThemeDefinition = getDefaultThemeDefinition();
+        }
+        return selectedThemeDefinition;
+    };
+
+    /**
+     * Builds a stylesheet model using the theme definition.
+     * @param {Object} themeDefinition
+     * @return {CSSStyleSheet}
+     */
+    const buildStylheetSheet = (themeDefinition) => {
+        const stylesheetContent = themeTag(themeDefinition);
+        const stylesheet = new CSSStyleSheet();
+        stylesheet.title = themeDefinition.name;
+        stylesheet.replaceSync(stylesheetContent);
+        return stylesheet;
+    };
+
+    /**
+     * Applies a theme with the provided theme name. The theme definition is obtained through the PluginRegistry and the
+     * theme definition is converted to a live stylesheet model which is then applied in the document.
+     * @param {string} themeName
+     */
+    const applyTheme = (themeName) => {
+      const themeDefinition = getThemeDefinitionByName(themeName);
+      const stylesheet = buildStylheetSheet(themeDefinition);
+      document.adoptedStyleSheets = [stylesheet];
+    };
+
+    const themeTag = (themeDefinition) => {
```

Comment:
Would it not be better to set default value if it is missing in the definition>

Reply:
I have described it in a google doc which I'm going to move in the dev guides